akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -287,15 +296,15 @@ private[akka] class RemoteWatcher(
    * does not exist.
    */
   def reWatch(address: Address): Unit =
-    watching.foreach {
-      case (wee: InternalActorRef, wer: InternalActorRef) ⇒
-        if (wee.path.address == address) {
-          // this re-watch will result in a RewatchRemote message to this actor
-          // must be a special message to be able to detect if an UnwatchRemote comes in
-          // before the extra RewatchRemote, then the re-watch should be ignored
-          log.debug("Re-watch [{} -> {}]", wer, wee)
-          wee.sendSystemMessage(new Rewatch(wee, wer)) // ➡➡➡ NEVER SEND THE SAME SYSTEM MESSAGE OBJECT TO TWO ACTORS ⬅⬅⬅
-        }
+    for {
+      (wee: InternalActorRef, wers) ← watching if wee.path.address == address
+      wer ← wers if wer.isInstanceOf[InternalActorRef]
```

Comment:
I would make that instanceOf into a `match` inside the next block, then isInstanceOf-asInstanceOf is not needed